src/twisted/conch/scripts/ckeygen.py

```diff
@@ -74,7 +74,12 @@ def run():
     if options['type']:
         if options['type'].lower() in supportedKeyTypes:
             print('Generating public/private %s key pair.' % (options['type']))
-            supportedKeyTypes[options['type'].lower()](options)
+            if options['type'].lower() == 'rsa':
+                generateRSAkey(options)
+            elif options['type'].lower() == 'dsa':
+                generateDSAkey(options)
+            else:
+                generateECDSAkey(options)
```

Comment:
Ok, so while it's nice that this improves the coverage, I'm a bit confused now. I think your decorator is wildly over-thinking things.

I think you can achieve the coverage you want by adding the decorator *wrapper* to the dictionary, rather than adding the raw function itself. That would let you keep using the dictionary here, but also fix your coverage problem.